xpublish_edr/geometry/common.py

```diff
@@ -19,3 +28,106 @@ def is_regular_xy_coords(ds: xr.Dataset) -> bool:
     Check if the dataset has 2D coordinates
     """
     return coord_is_regular(ds.cf["X"]) and coord_is_regular(ds.cf["Y"])
+
+
+def dataset_crs(ds: xr.Dataset) -> pyproj.CRS:
+    grid_mapping_names = ds.cf.grid_mapping_names
+    if len(grid_mapping_names) == 0:
```

Comment:
we may want to assert that 
```python
keys = ds.cf.keys()
assert "latitude" in ds.cf.keys() and "longitude" in ds.cf.keys()
```

Reply:
Yeah this is about finding the right default case